src/lib/npm.js

```diff
@@ -55,7 +55,7 @@ async function validateToken(otp, npm) {
 
 function getRegistry(pkg, conf) {
   if (pkg.publishConfig && pkg.publishConfig.registry) return pkg.publishConfig.registry;
-
+  if (pkg.name === undefined) throw new Error(`Your package doesn't have name`);
```

Comment:
```suggestion
   if (pkg.name === undefined) throw new Error(`package.json is missing "name" key`);
```

I'm not sure if this is the right place to throw the error, I'd throw it as soon as we read out `pkg`. The name property is not used in the `getRegistry` method

Reply:
the next line is using the name property `name[0]`, but I agree with you, it could be better if we placed it in the `index.js`.